internal/contour/handler.go

```diff
@@ -80,14 +82,34 @@ type opDelete struct {
 }
 
 func (e *EventHandler) OnAdd(obj interface{}) {
+	obj, err := e.UnstructuredConverter.Convert(obj)
```

Comment:
its sort of a bit inconsistent that we refer to an embedded field by its fully qualified name, yet two lines below a different embedded object is called without qualification. I think it might be easier on the reader if either both were referred to by their embedded field names, or both anonymously.